boa/src/builtins/string/mod.rs

```diff
@@ -789,15 +788,189 @@ impl String {
-                .unwrap_or_default()
+            StdString::from_utf16_lossy(
+                &this_str
+                    .encode_utf16()
+                    .skip(position + search_length)
+                    .collect::<Vec<u16>>()
+            )
         )
         .into())
     }
 
+    /// `22.1.3.18 String.prototype.replaceAll ( searchValue, replaceValue )`
+    ///
+    /// The replaceAll() method returns a new string with all matches of a pattern replaced by a replacement.
+    ///
+    /// The pattern can be a string or a RegExp, and the replacement can be a string or a function to be called for each match.
+    ///
+    /// The original string is left unchanged.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.replaceall
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace
+    pub(crate) fn replace_all(
+        this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> Result<JsValue> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = this.require_object_coercible(context)?;
+
+        let search_value = args.get(0).cloned().unwrap_or_default();
+        let replace_value = args.get(1).cloned().unwrap_or_default();
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null_or_undefined() {
+            // a. Let isRegExp be ? IsRegExp(searchValue).
+            if let Some(obj) = search_value.as_object() {
+                // b. If isRegExp is true, then
+                if obj.is_regexp() {
+                    // i. Let flags be ? Get(searchValue, "flags").
+                    let flags = obj.get("flags", context)?;
+
+                    // ii. Perform ? RequireObjectCoercible(flags).
+                    flags.require_object_coercible(context)?;
+
+                    // iii. If ? ToString(flags) does not contain "g", throw a TypeError exception.
+                    if !flags.to_string(context)?.contains('g') {
+                        return context.throw_type_error(
+                            "String.prototype.replaceAll called with a non-global RegExp argument",
+                        );
+                    }
+                }
+            }
+
+            // c. Let replacer be ? GetMethod(searchValue, @@replace).
+            let replacer = search_value
+                .as_object()
+                .unwrap_or_default()
+                .get_method(context, WellKnownSymbols::replace())?;
+
+            // d. If replacer is not undefined, then
+            if let Some(replacer) = replacer {
+                // i. Return ? Call(replacer, searchValue, « O, replaceValue »).
+                return context.call(&replacer.into(), &search_value, &[o.into(), replace_value]);
```

Comment:
```suggestion
                return replacer.call(&search_value, &[o.into(), replace_value], context);
```
It's better to call `.call` on `GcObject` so we avoid some checks and conversions

Reply:
Done :)